src/cascadia/TerminalSettingsEditor/ColorSchemes.cpp

```diff
@@ -175,41 +64,21 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
                                                    const SelectionChangedEventArgs& args)
     {
         //  Update the color scheme this page is modifying
-        const auto colorScheme{ args.AddedItems().GetAt(0).try_as<Model::ColorScheme>() };
-        CurrentColorScheme(colorScheme);
-        _UpdateColorTable(colorScheme);
-
-        _State.LastSelectedScheme(colorScheme.Name());
-
-        // Set the text disclaimer for the text box
-        hstring disclaimer{};
-        const std::wstring schemeName{ colorScheme.Name() };
-        if (std::find(std::begin(InBoxSchemes), std::end(InBoxSchemes), schemeName) != std::end(InBoxSchemes))
+        if (args.AddedItems().Size() > 0)
         {
-            // load disclaimer for in-box profiles
-            disclaimer = RS_(L"ColorScheme_DeleteButtonDisclaimerInBox");
-        }
-        DeleteButtonDisclaimer().Text(disclaimer);
+            const auto colorScheme{ args.AddedItems().GetAt(0).try_as<ColorSchemeViewModel>() };
+            _ViewModel.RequestSetCurrentScheme(*colorScheme);
```

Comment:
Why is it `RequestSetCurrentScheme(x)` and not `CurrentScheme(x)`? Can it _reject_ a request?

Reply:
For now, no, but I like having it this way to stress that ultimately it is the `ViewModel` that is doing the renaming, and the page is requesting for it. This is also consistent with our `AddProfile` page - `RequestAddNew` never gets rejected but I think it is a good name to indicate that who has the final say here